osu.Game.Tests/Visual/SongSelect/TestScenePlaySongSelect.cs

```diff
@@ -883,7 +886,7 @@ private void addManyTestMaps()
             {
                 var usableRulesets = rulesets.AvailableRulesets.Where(r => r.OnlineID != 2).ToArray();
 
-                for (int i = 0; i < 100; i += 10)
+                for (int i = 0; i < 10; i++)
                     manager.Import(TestResources.CreateTestBeatmapSetInfo(rulesets: usableRulesets)).Wait();
```

Comment:
Due to how this method works, this can still theoretically fail. If the beatmap count in the set is not provided, a random one is chosen:

https://github.com/peppy/osu/blob/5c8e317a6e7b647b40e1f92a9d0d6eb17dbafda5/osu.Game.Tests/Resources/TestResources.cs#L103-L104

While ruleset ID cycles across those beatmaps, ensuring relatively even distribution:

https://github.com/peppy/osu/blob/5c8e317a6e7b647b40e1f92a9d0d6eb17dbafda5/osu.Game.Tests/Resources/TestResources.cs#L82-L83

you could potentially choose all sets with only 1 beatmap and have one of the tests fixed above fail because of that (as i.e. one of them relies on having a taiko beatmap present).

Any objections to just fixing the difficulty count here to something like 10-12 to ensure this is not possible?

Reply:
but the rulesets aren't random and cycle sequentially right (and the count is global)? i think it should be fine?

Reply:
ah, it's not as global as i thought it was.

setting the difficulty count may be okay, but the reason it wasn't set was to fuzz for this specific test scene, i belive..

Reply:
In that case `addManyTestMaps` could receive an optional parameter with how many difficulties to create for each map, as both of the tests fixed above use that method anyhow.

Reply:
I think that sounds okay.

Reply:
Have made this change, would appreciate a second look to make sure it meets what you'd expect.